examples/asr/quartznet_speech_commands.py

```diff
@@ -0,0 +1,341 @@
+    )
+    parser.add_argument(
+        "--num_epochs", type=int, default=None, required=False, help="number of epochs to train",
+    )
+    parser.add_argument(
+        "--model_config", type=str, required=True, help="model configuration file: model.yaml",
+    )
+
+    # Create new args
+    parser.add_argument("--exp_name", default="Jasper_Speech_Commands", type=str)
+    parser.add_argument('--min_lr', default=1e-3, type=float)
+    parser.add_argument("--beta1", default=0.95, type=float)
+    parser.add_argument("--beta2", default=0.5, type=float)
+    parser.add_argument("--warmup_ratio", default=0.0, type=float)
+    parser.add_argument("--hold_ratio", default=0.0, type=float)
+    parser.add_argument(
+        "--load_dir", default=None, type=str, help="directory with pre-trained checkpoint",
+    )
+
+    args = parser.parse_args()
+
+    if args.max_steps is not None and args.num_epochs is not None:
+        raise ValueError("Either max_steps or num_epochs should be provided.")
+    return args
+
+
+def construct_name(name, lr, batch_size, max_steps, num_epochs, wd, optimizer, iter_per_step):
+    if max_steps is not None:
+        return "{0}-lr_{1}-bs_{2}-s_{3}-wd_{4}-opt_{5}-ips_{6}".format(
+            name, lr, batch_size, max_steps, wd, optimizer, iter_per_step
+        )
+    else:
+        return "{0}-lr_{1}-bs_{2}-e_{3}-wd_{4}-opt_{5}-ips_{6}".format(
+            name, lr, batch_size, num_epochs, wd, optimizer, iter_per_step
+        )
+
+
+def create_all_dags(args, neural_factory):
+    yaml = YAML(typ="safe")
+    with open(args.model_config) as f:
+        jasper_params = yaml.load(f)
+
+    labels = jasper_params['labels']  # Vocab of tokens
+    sample_rate = jasper_params['sample_rate']
+
+    # Calculate num_workers for dataloader
+    total_cpus = os.cpu_count()
+    cpu_per_traindl = max(int(total_cpus / neural_factory.world_size), 1)
+
+    # perturb_config = jasper_params.get('perturb', None)
+    train_dl_params = copy.deepcopy(jasper_params["AudioToSpeechLabelDataLayer"])
+    train_dl_params.update(jasper_params["AudioToSpeechLabelDataLayer"]["train"])
+    del train_dl_params["train"]
+    del train_dl_params["eval"]
+    # del train_dl_params["normalize_transcripts"]
+
+    # Look for augmentations
+    audio_augmentor = jasper_params.get('AudioAugmentor', None)
+
+    data_layer = nemo_asr.AudioToSpeechLabelDataLayer(
+        manifest_filepath=args.train_dataset,
+        labels=labels,
+        sample_rate=sample_rate,
+        batch_size=args.batch_size,
+        num_workers=cpu_per_traindl,
+        augmentor=audio_augmentor,
+        **train_dl_params,
+    )
+
+    crop_pad_augmentation = nemo_asr.CropOrPadSpectrogramAugmentation(audio_length=128)
+
+    N = len(data_layer)
+    steps_per_epoch = math.ceil(N / (args.batch_size * args.iter_per_step * args.num_gpus))
+    print("Steps per epoch :", steps_per_epoch)
```

Comment:
Shouldn't print statement here replaced by nemo.logging ?

Reply:
Fixed in latest commit.